Src/Base/AMReX_GpuLaunchFunctsG.H

```diff
@@ -997,6 +997,146 @@ void For (Box const& box1, T1 ncomp1, L1&& f1,
                 box3,ncomp3,std::forward<L3>(f3));
 }
 
+template <typename T, typename L, typename M=amrex::EnableIf_t<std::is_integral<T>::value> >
+void HostDeviceParallelFor (T n, L&& f) noexcept
+{
+    if (Gpu::inLaunchRegion()) {
+        ParallelFor(n,std::forward<L>(f));
+    } else {
+        for (T i = 0; i < n; ++i) f(i);
```

Comment:
since this is `ParallelFor`, do we want `AMREX_PRAGMA_SIMD` here for the CPU version?

Reply:
Yes, I will modify it.

Reply:
updated